numpy/core/src/umath/ufunc_object.c

```diff
@@ -6435,8 +6440,8 @@ ufunc_at(PyUFuncObject *ufunc, PyObject *args)
          */
         if ((ufuncimpl->contiguous_indexed_loop != NULL) &&
                 (PyArray_NDIM(op1_array) == 1)  &&
-                (ufuncimpl->nin == 1 || PyArray_NDIM(op2_array) == 1) && 
-                (iter->numiter == 1)) {
+                (op2_array != NULL && PyArray_NDIM(op2_array) == 1) &&
+                (iter->nd == 1)) {
```

Comment:
Jesh, this is crazy, but we have to worry about this?
```
In [1]: arr = np.zeros(5)
In [2]: np.add.at(arr, slice(None), np.ones(5))
```
so I suspect that we need to say that to check a bit more.  `->numiter` was maybe even right.  I will look up tomorrow what needs to be checked to make this work right...

Reply:
I think the `numiter` check was actually great and correct.  What we need is:
```
/* The index must be exactly one integer index array: */
iter->subspace_iter == NULL && iter->numiter == 1
```
we actually do not need `iter->nd`.  I.e. this is (maybe surprisingly) perfectly fine:
```
np.exp.at(np.arange(5), np.ones((5, 5), dtype=np.intp))
```

We do still need the `PyArray_NDIM(op2_array) == 1` because while that could be generalized it would need care to make sure the iteration order of the index and `op2` match for higher dimensions.

Reply:
Thanks for digging in. Adopted in 163d3c5

Reply:
Cool, could you add the test?
```
arr = np.zeros(5)
np.add.at(arr, slice(None), np.ones(5))
assert_array_equal(arr, np.ones(5))
```